src/elexmodel/models/BaseElectionModel.py

```diff
@@ -264,3 +266,37 @@ def get_coefficients(self):
         These coefficients are for the point prediciton only, not for the lower or upper intervals models.
         """
         return self.features_to_coefficients
+
+    def find_optimal_lambda(
+        self,
+        reporting_units,
+        possible_lambda_values=[],
+        estimand="",
+        K=3,
+    ):
+        if len(possible_lambda_values) == 0:
+            return 0, None
+        if len(possible_lambda_values) == 1:
+            return possible_lambda_values[0], None
+
+        MAPE_arr = np.zeros_like(possible_lambda_values)  # array of MAPE sums for each lambda input
+
+        # loop through each lambda
+        for loc, lam in enumerate(possible_lambda_values):
+            for train_index, test_index in math_utils.get_kfold_splits(reporting_units, K):
+                # build model with custom lambda
+                train = reporting_units.iloc[train_index]
```

Comment:
I think we will want to randomize the order of the reporting units before splitting to make sure that our MAPE value isn't dependent on the ordering that we get